src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/MercuryArticlePageObject.java

```diff
@@ -157,14 +186,126 @@ public void verifyCanonicalUrl() {
+    Assertion
+        .assertTrue(
+            numberOfReplies == commentsRepliesList.get(index).findElements(By.cssSelector("li"))
+                .size(), "Replies counter doesn't work");
+  }
+
+  public void verifyTapOnUserRedirectToUserPage(int index) {
+    String userName = commentsUsernames.get(index).getText();
+    commentsUsernames.get(index).click();
+    String subUrl =
+        driver.getCurrentUrl().substring(driver.getCurrentUrl().indexOf("/wiki/") + 6,
+            driver.getCurrentUrl().length());
+    Assertion.assertTrue(subUrl.equals("User:" + userName), "Url doesn't contain user page");
+  }
+
+  public void verifyCommentsCounterIsCorrect() {
+    int numberOfComments =
+        Integer.parseInt(commentsHeader.getText().substring(0,
+            commentsHeader.getText().indexOf(" ")));
+    while (nextCommentPageButton.isDisplayed()) {
+      numberOfComments -= allComments.size();
+      nextCommentPageButton.click();
+    }
+    numberOfComments -= allComments.size();
+    Assertion.assertTrue(numberOfComments == 0, "There are "
+        + numberOfComments + " untracked comments");
+  }
+
+  public void verifyMediaInComments(String mediaType, int index) {
+    WebElement mediaInComment;
+    String methodName = "verifyMediaInComments";
+    try {
+      if (mediaType.equals("Video")) {
+        mediaInComment = allComments.get(index).findElement(By.cssSelector("figure.comment-video"));
+      } else {
+        mediaInComment = allComments.get(index).findElement(By.cssSelector("figure"));
+      }
+      Assertion.assertTrue(mediaInComment.findElement(By.cssSelector("img")).isDisplayed(),
+          mediaType + " thumbnail isn't displayed");
+      Assertion.assertTrue(mediaInComment.findElement(By.cssSelector("a")).getAttribute("href")
+          .contains("/wiki/File:"), mediaType + " anchor isn't displayed");
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.log(methodName, "There is no " + mediaType + " in that comment", false);
+      PageObjectLogging.log(methodName, e.getMessage(), false);
+    }
+  }
+
+  public void verifyChevronRotatesWhenTapped() {
+    boolean collapsed = showCommentsButton.getAttribute("class").contains("collapsed");
+    clickCommentsHeader();
+    Assertion.assertFalse(
+        collapsed == showCommentsButton.getAttribute("class").contains("collapsed"),
+        "Chevron didn't turn");
+  }
+
+  public void verifyFooterElements() {
+    String methodName = "verifyFooterElements";
+    int elementCounter = 0;
+    try {
+      waitForElementByElement(footerLogo);
+      Assertion.assertTrue(footerLogo.isDisplayed(), "Wikia logo isn't displayed");
+      for (WebElement element : footerLinks) {
+        Assertion.assertTrue(element.isDisplayed(), element.getText() + " isn't displayed");
+        ++elementCounter;
+      }
+      Assertion.assertTrue(elementCounter == 11, "Some elements aren't displayed");
```

Comment:
Please use a meaningful variable/const to represents '11'. It's unclear what the significant of 11 is in this assertion.